yt_dlp/extractor/douyutv.py

```diff
@@ -1,31 +1,62 @@
 import time
 import hashlib
-import re
 import urllib
+import uuid
 
 from .common import InfoExtractor
+from .openload import PhantomJSwrapper
 from ..utils import (
-    ExtractorError,
+    UserNotLive,
+    int_or_none,
+    str_or_none,
+    url_or_none,
+    traverse_obj,
+    determine_ext,
+    parse_resolution,
+    urlencode_postdata,
     unescapeHTML,
-    unified_strdate,
+    js_to_json,
     urljoin,
 )
 
 
-class DouyuTVIE(InfoExtractor):
-    IE_DESC = '斗鱼'
+class DouyuBaseIE(InfoExtractor):
+    _cryptojs_md5 = None
+
+    def _get_cryptojs_md5(self, video_id):
+        for url in [
+            'https://cdnjs.cloudflare.com/ajax/libs/crypto-js/3.1.2/rollups/md5.js',
+            'https://cdn.bootcdn.net/ajax/libs/crypto-js/3.1.2/rollups/md5.js',
+        ]:
+            if DouyuBaseIE._cryptojs_md5:
+                break
+            DouyuBaseIE._cryptojs_md5 = self._download_webpage(
+                url, video_id, note='Downloading signing dependency', fatal=False)
+        return DouyuBaseIE._cryptojs_md5
```

Comment:
Better to cache on disk. See https://github.com/yt-dlp/yt-dlp/blob/a250b247334ce9f641e709cbb64974da6034a2b3/yt_dlp/extractor/youtube.py#L3124-L3127

https://github.com/yt-dlp/yt-dlp/blob/a250b247334ce9f641e709cbb64974da6034a2b3/yt_dlp/extractor/iqiyi.py#L524-L544

Reply:
Thanks for the references. Changed it to cache on disk